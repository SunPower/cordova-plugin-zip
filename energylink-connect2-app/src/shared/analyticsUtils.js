import {
  __,
  compose,
  curry,
  gt,
  isEmpty,
  join,
  lensIndex,
  map,
  not,
  over,
  props,
  propSatisfies,
  toLower,
  toUpper,
  when
} from 'ramda'

export const isNotEmpty = compose(not, isEmpty)

export const capitalizeWord = when(
  propSatisfies(gt(__, 0), 'length'),
  compose(join(''), over(lensIndex(0), toUpper), toLower)
)

const stringValueOrEmptyString = stringValue => stringValue || ''

const getStringProps = curry(compose(map(stringValueOrEmptyString), props))

export const getUserProfile = getStringProps([
  'uniqueId',
  'firstName',
  'lastName',
  'email',
  'dealerName',
  'recordType'
])
