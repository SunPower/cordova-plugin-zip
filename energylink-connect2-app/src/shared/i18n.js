import { useMemo } from 'react'
import { useSelector } from 'react-redux'

import * as languages from './locales'
import * as localizedComponents from './localizedComponents'

const getTranslate = translations => (key, ...params) => {
  let value = translations[key] || key
  if (params) {
    params.forEach((p, idx) => {
      value = value.replace(`{${idx}}`, p)
    })
  }
  return value
}

export const useI18n = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language
  return useMemo(() => getTranslate(languages[locale]), [locale])
}

export const useI18nComponent = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language || {}

  return useMemo(
    () => componentName => {
      const path = `${locale}_${componentName}`
      const Component = localizedComponents[path] || {}
      if (typeof Component !== 'function') {
        throw new Error(`Component not found: ${path}`)
      }
      return Component()
    },
    [locale]
  )
}
