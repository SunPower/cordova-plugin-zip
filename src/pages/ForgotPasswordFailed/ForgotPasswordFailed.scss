@import "../../styles/variables";

.forgot-password {
  .container {
    flex-grow: 0;
    &:last-child {
      flex-grow: 1;
      margin-left: 0.5em;
      margin-right: 0.5em;
    }

    .title {
      h1 {
        font-weight: $bold-text;
        font-size: $font-size;
        line-height: 1rem;
        text-align: center;
        letter-spacing: $letter-spacing-big;
        text-transform: uppercase;
      }
    }

    .subtitle {
      h2 {
        width: 90%;
        margin: 0 auto;
      }
    }

    .link {
      font-size: $font-size;
      line-height: 1rem;
      text-align: center;
      letter-spacing: $letter-spacing-default;
      text-transform: uppercase;
    }
  }
}
