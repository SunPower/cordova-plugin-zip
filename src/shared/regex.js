export const PASSWORD_REGEXP = /^(?=.*[a-z])(?=.*\d)[A-Za-z\d]{8,}$/
export const EMAIL_REGEXP = /^\S+@\S+\.\S+$/
export const PHONE_REGEXP = /^(1\s?)?((([0-9]{3}))|[0-9]{3})[\s-]?[0-9]{3}[\s-]?[0-9]{4}$/
