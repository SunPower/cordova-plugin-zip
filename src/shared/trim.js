export const trimObject = obj => {
  return Object.entries(obj)
    .map(([k, v]) => ({
      [k]: typeof v === 'string' || v instanceof String ? v.trim() : v
    }))
    .reduce((acc, kv) => ({ ...acc, ...kv }), {})
}
