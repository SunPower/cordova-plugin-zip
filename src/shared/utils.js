export const either = (condition, whenTrue, whenFalse) =>
  condition ? whenTrue : whenFalse
