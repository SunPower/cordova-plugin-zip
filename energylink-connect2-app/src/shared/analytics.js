import appVersion from '../macros/appVersion.macro'
import { capitalizeWord, getUserProfile } from 'shared/analyticsUtils'
import { MIXPANEL_EVENT_QUEUED } from 'state/actions/analytics'
import { getAppleDeviceFamily } from 'shared/appleDevicesTable'
import { prop } from 'ramda'

const getAppAndDeviceProperties = () => {
  const { device } = window
  const { model, version, platform, manufacturer } = device

  let deviceModel
  let deviceBrowserVersion

  if (platform === 'iOS') {
    deviceModel = getAppleDeviceFamily(model)
    deviceBrowserVersion = version
  } else {
    deviceModel = `${manufacturer}, ${model}`
  }

  return {
    'App Build': appVersion(),
    'App Flavor': process.env.REACT_APP_FLAVOR,
    'OS Version': version,
    ...(deviceBrowserVersion ? { $browser_version: deviceBrowserVersion } : {}),
    $device: deviceModel
  }
}

export const loggedIn = user => {
  const { mixpanel } = window
  const {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  } = getUserProfile(user)
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Login', { Success: true })

  return MIXPANEL_EVENT_QUEUED('Login - Success')
}

export const loginFailed = () => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.track('Login', { Success: false })

  return MIXPANEL_EVENT_QUEUED('Login - Failed')
}

export const scanPVS = ({ pvsSN, entryMethod }, event) => {
  const { mixpanel } = window

  mixpanel.register({ 'PVS SN': pvsSN })
  mixpanel.track('Scan PVS Tag', {
    'Entry Method': entryMethod
  })

  return MIXPANEL_EVENT_QUEUED(event)
}

export const siteNotFound = ({
  uniqueId: userId,
  firstName,
  lastName,
  email,
  dealerName,
  recordType: dealerType
}) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', { Found: false })

  return MIXPANEL_EVENT_QUEUED('Find Site - site not found')
}

export const siteFound = (
  {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  },
  { city, st_id, postalCode, siteKey, commissioned }
) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', {
    Found: true,
    ...{
      $city: city,
      State: st_id,
      'Zip Code': postalCode,
      'Site ID': siteKey,
      Commissioned: commissioned
    }
  })

  return MIXPANEL_EVENT_QUEUED('Find Site - site found')
}

export const saveConfiguration = config => {
  const { mixpanel } = window
  mixpanel.track('Configure', config)
  return MIXPANEL_EVENT_QUEUED('Configure - submit site config')
}

export const saveInventory = (inventory, hasEss) => {
  const { mixpanel } = window

  mixpanel.register({
    'Equinox Storage Site': hasEss
  })

  mixpanel.track('Inventory', {
    'AC Module Count': prop('AC_MODULES', inventory),
    'ESS System Size': hasEss ? prop('ESS', inventory) : 'None',
    'Equinox Storage Site': hasEss
  })
  return MIXPANEL_EVENT_QUEUED('Inventory')
}

export const deviceResume = pageResume => {
  const { mixpanel } = window
  mixpanel.register({
    Page: pageResume
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

/**
 *
 * @param {string}[interfaceType] - The type of connection the user has. either can be  PVS, Cell, or Customer WiFi
 * @return {Action<null, null>}
 */
export const internetConnection = (interfaceType = '') => {
  const { mixpanel } = window
  mixpanel.register({
    'Internet Connection': interfaceType
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

export const firmwareUpdateEvent = data => {
  const { mixpanel } = window
  mixpanel.track('Firmware Update', {
    'Current FW Version': prop('fromFWVersion', data),
    'Current BN Version': prop('fromBuildNumber', data),
    'New FW Version': prop('toFWVersion', data),
    'New BN Version': prop('toBuildNumber', data),
    Success: prop('success', data)
  })
  return MIXPANEL_EVENT_QUEUED('Firmware update - Update successful')
}

export const timeMixPanelEvent = name => {
  if (!name) throw new Error('The name parameter is required')
  const { mixpanel } = window
  mixpanel.time_event(name)
}

export const commissionSite = ({ duration, timeConfiguring }) => {
  const { mixpanel } = window
  mixpanel.track('Commission Site', {
    $duration: duration,
    'Time Configuring': timeConfiguring
  })
  return MIXPANEL_EVENT_QUEUED(
    'Commission Site - first successful site configuration'
  )
}
