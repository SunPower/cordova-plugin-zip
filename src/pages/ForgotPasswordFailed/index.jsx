import React from 'react'
import { Redirect, Link } from 'react-router-dom'
import paths from '../Router/paths'
import Logo from '../../components/Logo'
import { useI18n } from '../../shared/i18n'

import './ForgotPasswordFailed.scss'

function ForgotPasswordFailed({ isLoggedIn = false }) {
  const t = useI18n()

  if (isLoggedIn) {
    return <Redirect to={paths.ROOT} />
  }

  return (
    <section className="forgot-password section full-min-height is-flex">
      <div className="container mt-50 mb-100">
        <Logo />
        <div className="container is-spaced has-text-centered mt-40 title is-6">
          <h1>{t('FORGOT_PASSWORD_FAILED')}</h1>
        </div>
        <div className="container is-spaced mt-30 has-text-centered subtitle is-6">
          <h2>{t('FORGOT_PASSWORD_FAILED_TEXT')}</h2>
        </div>
      </div>
      <div className="container is-spaced mt-100 has-text-centered">
        <Link className="link" to={{ pathname: paths.SIGNUP }}>
          {t('TAB_TITLE_SIGNUP')}
        </Link>
      </div>
    </section>
  )
}

export default ForgotPasswordFailed
