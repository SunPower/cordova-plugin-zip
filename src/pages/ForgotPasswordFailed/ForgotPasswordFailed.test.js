import React from 'react'
import { shallow } from 'enzyme'
import ForgotPasswordFailed from '.'
import * as i18n from '../../shared/i18n'

describe('ForgotPasswordFailed page', () => {
  beforeEach(() => {
    jest
      .spyOn(i18n, 'useI18n')
      .mockImplementation(path => (key, ...params) =>
        `${key.toUpperCase()} ${params.join('_')}`.trim()
      )
  })

  it('renders without crashing', () => {
    const component = shallow(<ForgotPasswordFailed />)
    expect(component).toMatchSnapshot()
  })

  it('redirects if we are logged in', () => {
    const component = shallow(<ForgotPasswordFailed isLoggedIn="true" />)
    expect(component).toMatchSnapshot()
  })
})
