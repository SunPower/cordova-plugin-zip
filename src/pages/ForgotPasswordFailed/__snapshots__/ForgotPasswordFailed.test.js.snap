// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordFailed page redirects if we are logged in 1`] = `
<Redirect
  to="/"
/>
`;

exports[`ForgotPasswordFailed page renders without crashing 1`] = `
<section
  className="forgot-password section full-min-height is-flex"
>
  <div
    className="container mt-50 mb-100"
  >
    <Logo />
    <div
      className="container is-spaced has-text-centered mt-40 title is-6"
    >
      <h1>
        FORGOT_PASSWORD_FAILED
      </h1>
    </div>
    <div
      className="container is-spaced mt-30 has-text-centered subtitle is-6"
    >
      <h2>
        FORGOT_PASSWORD_FAILED_TEXT
      </h2>
    </div>
  </div>
  <div
    className="container is-spaced mt-100 has-text-centered"
  >
    <Link
      className="link"
      to={
        Object {
          "pathname": "/sign-up",
        }
      }
    >
      TAB_TITLE_SIGNUP
    </Link>
  </div>
</section>
`;
